Source: written by Tim Kosse
Upstream: yes, bug #12930, revision 10966
Reason: Fix build compatibility with libfilezilla 0.43.0

--- FileZilla3/trunk/src/engine/sftp/sftpcontrolsocket.cpp	2023/05/16 09:15:51	10965
+++ FileZilla3/trunk/src/engine/sftp/sftpcontrolsocket.cpp	2023/05/16 09:26:24	10966
@@ -511,17 +511,14 @@
 	if (input_parser_) {
 		input_parser_.reset();
 
-		auto threadEventsFilter = [&](fz::event_loop::Events::value_type const& ev) -> bool {
-			if (ev.first != this) {
-				return false;
-			}
-			else if (ev.second->derived_type() == CSftpEvent::type() || ev.second->derived_type() == CSftpListEvent::type()) {
+		auto threadEventsFilter = [](fz::event_base const& ev) -> bool {
+			if (ev.derived_type() == CSftpEvent::type() || ev.derived_type() == CSftpListEvent::type()) {
 				return true;
 			}
 			return false;
 		};
 
-		event_loop_.filter_events(threadEventsFilter);
+		filter_events(threadEventsFilter);
 	}
 	process_.reset();
 
--- FileZilla3/trunk/src/engine/storj/storjcontrolsocket.cpp	2023/05/16 09:15:51	10965
+++ FileZilla3/trunk/src/engine/storj/storjcontrolsocket.cpp	2023/05/16 09:26:24	10966
@@ -336,17 +336,14 @@
 	if (input_thread_) {
 		input_thread_.reset();
 
-		auto threadEventsFilter = [&](fz::event_loop::Events::value_type const& ev) -> bool {
-			if (ev.first != this) {
-				return false;
-			}
-			else if (ev.second->derived_type() == CStorjEvent::type() || ev.second->derived_type() == StorjTerminateEvent::type()) {
+		auto threadEventsFilter = [](fz::event_base const& ev) -> bool {
+			if (ev.derived_type() == CStorjEvent::type() || ev.derived_type() == StorjTerminateEvent::type()) {
 				return true;
 			}
 			return false;
 		};
 
-		event_loop_.filter_events(threadEventsFilter);
+		filter_events(threadEventsFilter);
 	}
 	process_.reset();
 
